crates/wasm-smith/src/core/code_builder.rs

```diff
@@ -1684,6 +1684,18 @@ fn have_memory(module: &Module, _: &mut CodeBuilder) -> bool {
     module.memories.len() > 0
 }
 
+#[inline]
+fn float_enabled(module: &Module, _: &mut CodeBuilder) -> bool {
+    module.config.float_enabled()
+}
+
+#[inline]
+fn float_memory(module: &Module, builder: &mut CodeBuilder) -> bool {
+    module.config.float_enabled()
+        && ((builder.allocs.memory32.len() > 0 && builder.type_on_stack(ValType::I32))
+            || (builder.allocs.memory64.len() > 0 && builder.type_on_stack(ValType::I64)))
```

Comment:
did we expect here `ValType::F32`/`ValType::F64`?

Reply:
No, we expect here addr value which must be from i32/i64 register